Guides for OCI and AWS Metrics

n0tl3ss opened the pull request:
Preview:
[Collect Metrics with Micronaut and Monitor them on the Oracle Cloud.pdf](https://github.com/micronaut-projects/micronaut-guides/files/9281311/Collect.Metrics.with.Micronaut.and.Monitor.them.on.the.Oracle.Cloud.pdf)

n0tl3ss committed:
Draft guide for OCI Metrics

Commit:
Removed features from guides, since they are now in starter. Updated metadata.

n0tl3ss committed:
add aws cloudwatch metrics guide

n0tl3ss commented:
AWS Cloudwatch guide: [Collect Metrics with Micronaut and Monitor them on the AWS Cloudwatch.pdf](https://github.com/micronaut-projects/micronaut-guides/files/9566298/Collect.Metrics.with.Micronaut.and.Monitor.them.on.the.AWS.Cloudwatch.pdf)

burtbeckwith commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+external:micronaut-metrics/start.adoc[]
+
+common:requirements.adoc[]
```
remove this, it's in micronaut-metrics/start.adoc and getting rendered twice

burtbeckwith commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+
+external:micronaut-metrics/create-app.adoc[]
+
+== Monitor metrics on the AWS Cloudwatch
```
remove "the"

burtbeckwith commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+
+== Monitor metrics on the AWS Cloudwatch
+
+For publishing metrics on the AWS Cloudwatch you have to adjust your application configuration slightly. With that configuration change you should be able to create queries and monitor your metrics on the https://console.aws.amazon.com/cloudwatch#metricsV2:graph=~()[Cloudwatch].
```
```suggestion
To publish metrics to AWS Cloudwatch, you have to adjust your application configuration slightly, and then you will be able to create queries and monitor your metrics on https://console.aws.amazon.com/cloudwatch#metricsV2:graph=~()[Cloudwatch].
```

burtbeckwith commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+
+=== Visit the https://console.aws.amazon.com/cloudwatch#metricsV2:graph=~()[AWS Cloudwatch] Metrics.
+
+In the Metrics panel select your namespace that you choose earlier. We will select `micronaut_guide`.
```
choose -> chose

burtbeckwith commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+
+{empty} +
+
+Here you can browse different metrics and explore their values. In this example we want to check the price of bitcoin. We will select `Metrics with no dimensions`.
```
```suggestion
Here you can browse different metrics and explore their values. In this example we want to check the price of Bitcoin. Select `Metrics with no dimensions`.
```

burtbeckwith commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+
+{empty} +
+
+In this panel you can see our custom metrics about bitcoin. We can select (#2) `bitcoin.price.latest.value` to see on (#1) the graph how price of the bitcoin were changing through time.
```
```suggestion
In this panel you can see our custom metrics about Bitcoin. Select (#2) `bitcoin.price.latest.value` to see on (#1) the graph how the price of Bitcoin changed over time.
```

burtbeckwith commented on guides/micronaut-metrics-aws/metadata.json:
```diff
@@ -0,0 +1,13 @@
+{
+  "base": "micronaut-metrics",
+  "title": "Collect Metrics with Micronaut and Monitor them on the AWS Cloudwatch",
+  "intro": "Learn how to collect standard and custom metrics with the Micronaut framework and monitor them on the AWS Cloudwatch.",
```
```suggestion
  "title": "Collect Metrics with Micronaut and Monitor them on AWS Cloudwatch",
  "intro": "Learn how to collect standard and custom metrics with the Micronaut framework and monitor them on AWS Cloudwatch.",
```

burtbeckwith commented on guides/micronaut-metrics-oci/metadata.json:
```diff
@@ -0,0 +1,13 @@
+{
+  "base": "micronaut-metrics",
+  "title": "Collect Metrics with Micronaut and Monitor them on the Oracle Cloud",
+  "intro": "Learn how to collect standard and custom metrics with the Micronaut framework and monitor them on the Oracle Cloud.",
```
```suggestion
  "title": "Collect Metrics with Micronaut and Monitor them on Oracle Cloud",
  "intro": "Learn how to collect standard and custom metrics with the Micronaut framework and monitor them on Oracle Cloud.",
```

burtbeckwith commented on guides/micronaut-metrics-oci/micronaut-metrics-oci.adoc:
```diff
@@ -0,0 +1,32 @@
+
+== Monitor metrics on Oracle Cloud
+
+For publishing metrics on the Oracle Cloud you have to adjust your application configuration slightly. With that configuration change you should be able to create queries and monitor your metrics on the Oracle Cloud https://cloud.oracle.com/monitoring/explore[Metrics Explorer].
```
```suggestion
To publish metrics to Oracle Cloud, you have to adjust your application configuration slightly, and then you will be able to create queries and monitor your metrics on the Oracle Cloud https://cloud.oracle.com/monitoring/explore[Metrics Explorer].
```

burtbeckwith commented on guides/micronaut-metrics-oci/micronaut-metrics-oci.adoc:
```diff
@@ -0,0 +1,32 @@
+
+image::oci-metrics/query-editor.png[]
+
+After clicking on the (#4) Update Chart button you should be able to see values of your selected metric in the Metrics Explorer.
```
```suggestion
After clicking the (#4) Update Chart button you will see values of your selected metric in the Metrics Explorer.
```

n0tl3ss committed:
Update guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc

Co-authored-by: Burt Beckwith <burt@burtbeckwith.com>

n0tl3ss committed:
Update guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc

Co-authored-by: Burt Beckwith <burt@burtbeckwith.com>

n0tl3ss committed:
Update guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc

Co-authored-by: Burt Beckwith <burt@burtbeckwith.com>

n0tl3ss committed:
Update guides/micronaut-metrics-oci/metadata.json

Co-authored-by: Burt Beckwith <burt@burtbeckwith.com>

n0tl3ss committed:
Update guides/micronaut-metrics-oci/micronaut-metrics-oci.adoc

Co-authored-by: Burt Beckwith <burt@burtbeckwith.com>

n0tl3ss committed:
Update guides/micronaut-metrics-oci/micronaut-metrics-oci.adoc

Co-authored-by: Burt Beckwith <burt@burtbeckwith.com>

n0tl3ss committed:
Update guides/micronaut-metrics-aws/metadata.json

Co-authored-by: Burt Beckwith <burt@burtbeckwith.com>

n0tl3ss committed:
add suggeted changes and make guides smaller

n0tl3ss commented:
New version of guides (shortened): 

[Collect Metrics with Micronaut and Monitor them on AWS Cloudwatch.pdf](https://github.com/micronaut-projects/micronaut-guides/files/9584248/Collect.Metrics.with.Micronaut.and.Monitor.them.on.AWS.Cloudwatch.pdf)
[Collect Metrics with Micronaut and Monitor them on Oracle Cloud.pdf](https://github.com/micronaut-projects/micronaut-guides/files/9584249/Collect.Metrics.with.Micronaut.and.Monitor.them.on.Oracle.Cloud.pdf)

n0tl3ss committed:
fix authors

ghost commented:
> AWS Cloudwatch guide: [Collect Metrics with Micronaut and Monitor them on the AWS Cloudwatch.pdf](https://github.com/micronaut-projects/micronaut-guides/files/9566298/Collect.Metrics.with.Micronaut.and.Monitor.them.on.the.AWS.Cloudwatch.pdf)

There's no mention in the "What you will need" about requiring an AWS account.

n0tl3ss commented:
> > AWS Cloudwatch guide: [Collect Metrics with Micronaut and Monitor them on the AWS Cloudwatch.pdf](https://github.com/micronaut-projects/micronaut-guides/files/9566298/Collect.Metrics.with.Micronaut.and.Monitor.them.on.the.AWS.Cloudwatch.pdf)
> 
> There's no mention in the "What you will need" about requiring an AWS account.

I didn't see in any of AWS guide that is mentioned that AWS account is required.

n0tl3ss committed:
add aws common

n0tl3ss commented:
[Collect Metrics with Micronaut and Monitor them on AWS Cloudwatch.pdf](https://github.com/micronaut-projects/micronaut-guides/files/9586717/Collect.Metrics.with.Micronaut.and.Monitor.them.on.AWS.Cloudwatch.pdf)

shingjo commented on guides/micronaut-metrics-oci/src/main/resources/application.yml:
```diff
@@ -0,0 +1,58 @@
+    export:
+      oraclecloud:
+        enabled: true
+        namespace: change-me # <3>
```
Why have `change-me` instead of `micronaut_guide`?

sdelamo replied:
I have changed this here: https://github.com/micronaut-projects/micronaut-guides/pull/1042/commits/b3155533a894a7bc3a3e31485b9c74aa8cab2604

shingjo commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+external:micronaut-metrics/start.adoc[]
+
+common:aws.adoc[]
```
FYI: If you want the What you will need section to look the same as OCI see https://github.com/micronaut-projects/micronaut-guides/blob/master/guides/micronaut-cloud-database-aws/micronaut-cloud-database-aws.adoc

sdelamo commented on guides/micronaut-metrics-aws/metadata.json:
```diff
@@ -0,0 +1,13 @@
+{
+  "base": "micronaut-metrics",
+  "title": "Collect Metrics with Micronaut and Monitor them on AWS Cloudwatch",
```
```suggestion
  "title": "Collect Metrics with the Micronaut Framework and Monitor them on AWS Cloudwatch",
```

see: https://micronaut.io/brand-guidelines/

sdelamo commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+
+<1> Change to `false` to disable all metrics, e.g., per-environment
+<2> These default to `true` and are only here for convenience, to be able to disable a subset of metrics
+<3> Change namespace to something meaningful, for an example app name. In this guide we are using `micronaut_guide`
```
```suggestion
<3> Change the namespace to something meaningful, for example, the app's name. In this guide, we are using `micronautguide`
```

We alway use `micronautguide` wtithout underscore for the guide application name. Is there a reason why you used `micronaut_guide`?

sdelamo commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+
+=== Visit the https://console.aws.amazon.com/cloudwatch#metricsV2:graph=~()[AWS Cloudwatch] Metrics.
+
+In the Metrics panel select your namespace that you chose earlier. We will select `micronaut_guide`.
```
```suggestion
In the Metrics panel, select the namespace that you chose earlier. We select `micronautguide`.
```

We alway use micronautguide wtithout underscore for the guide application name. Is there a reason why you used `micronaut_guide`?

n0tl3ss replied:
There is none

sdelamo commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+
+{empty} +
+
+Here you can browse different metrics and explore their values. In this example we want to check the price of Bitcoin. Select `Metrics with no dimensions`.
```
```suggestion
Here you can browse different metrics and explore their values. In this example, we want to check the price of Bitcoin. Select `Metrics with no dimensions`.
```

sdelamo commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+
+{empty} +
+
+In this panel you can see our custom metrics about Bitcoin. Select (#2) `bitcoin.price.latest.value` to see on (#1) the graph how the price of Bitcoin changed over time.
```
```suggestion
In this panel, you can see our custom metrics about Bitcoin. Select (#2) `bitcoin.price.latest.value` to see on (#1) the graph how the price of Bitcoin changed over time.
```

sdelamo commented on guides/micronaut-metrics-aws/src/main/resources/application.yml:
```diff
@@ -0,0 +1,58 @@
+  http:
+    services:
+      kucoin: # <1>
+        urls:
```
you can use `url` instead of `urls` with a list. see: https://docs.micronaut.io/latest/guide/configurationreference.html#io.micronaut.http.client.ServiceHttpClientConfiguration

sdelamo commented on guides/micronaut-metrics-aws/src/main/resources/application.yml:
```diff
@@ -0,0 +1,58 @@
+    export:
+      cloudwatch:
+        enabled: true
+        namespace: change-me # <3>
```
```suggestion
        namespace: micronautguide # <3>
```

sdelamo commented on guides/micronaut-metrics-oci/metadata.json:
```diff
@@ -0,0 +1,13 @@
+{
+  "base": "micronaut-metrics",
+  "title": "Collect Metrics with Micronaut and Monitor them on Oracle Cloud",
```
```suggestion
  "title": "Collect Metrics with the Micronaut Framework and Monitor them on Oracle Cloud",
```

see https://micronaut.io/brand-guidelines/

sdelamo commented on guides/micronaut-metrics-aws/src/test/resources/application-test.yml:
```diff
@@ -0,0 +1,7 @@
+crypto:
+  initialDelay: 10h # <1>
```
```suggestion
  initialDelay: 10h
```
We don't show the file in the guide. We don't need a callout.

sdelamo commented on guides/micronaut-metrics-oci/micronaut-metrics-oci.adoc:
```diff
@@ -0,0 +1,32 @@
+
+<1> Change to `false` to disable all metrics, e.g., per-environment
+<2> These default to `true` and are only here for convenience, to be able to disable a subset of metrics
+<3> Change namespace to something meaningful, for an example app name. In this guide we are using `micronaut_guide`
```
you can extract a callout and reuse it in both guides.

sdelamo commented on guides/micronaut-metrics-oci/micronaut-metrics-oci.adoc:
```diff
@@ -0,0 +1,32 @@
+resource:application.yml[tag=metrics]
+
+<1> Change to `false` to disable all metrics, e.g., per-environment
+<2> These default to `true` and are only here for convenience, to be able to disable a subset of metrics
```
```suggestion
<2> These binders default to `true`. We list them for convenience to encourage you to experiment with disabling some of them. 
```

You could extract a callout and reuse it between both guides.

sdelamo commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+resource:application.yml[tag=metrics]
+
+<1> Change to `false` to disable all metrics, e.g., per-environment
+<2> These default to `true` and are only here for convenience, to be able to disable a subset of metrics
```
```suggestion
<2> These binders default to `true`. We list them for convenience to encourage you to experiment with disabling some of them. 
```

sdelamo commented on guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc:
```diff
@@ -0,0 +1,43 @@
+
+resource:application.yml[tag=metrics]
+
+<1> Change to `false` to disable all metrics, e.g., per-environment
```
you could extract a callout to avoid repetition between both guides.

sdelamo commented on guides/micronaut-metrics-oci/micronaut-metrics-oci.adoc:
```diff
@@ -0,0 +1,32 @@
+
+=== Visit the Oracle Cloud https://cloud.oracle.com/monitoring/explore[Metrics Explorer].
+
+In the query editor create the query that you want to monitor. You have to specify the compartment (#1), namespace (#2) and the name of the metric (#3).
```
```suggestion
In the query editor, create the query that you want to monitor. You have to specify the compartment (#1), namespace (#2), and the name of the metric (#3).
```

sdelamo commented on guides/micronaut-metrics-oci/micronaut-metrics-oci.adoc:
```diff
@@ -0,0 +1,32 @@
+
+image::oci-metrics/query-editor.png[]
+
+After clicking the (#4) Update Chart button you will see values of your selected metric in the Metrics Explorer.
```
```suggestion
After clicking the (#4) Update Chart button, you will see the values of your selected metric in the Metrics Explorer.
```

sdelamo commented on guides/micronaut-metrics-oci/src/main/resources/application.yml:
```diff
@@ -0,0 +1,58 @@
+    export:
+      oraclecloud:
+        enabled: true
+        namespace: change-me # <3>
```
```suggestion
        namespace: micronautguide # <3>
```

sdelamo commented on guides/micronaut-metrics-oci/src/main/resources/application.yml:
```diff
@@ -0,0 +1,58 @@
+    driverClassName: org.h2.Driver
+    username: sa
+    password: ''
+    schema-generate: CREATE_DROP
```
since we use flyway, this should be: 
```suggestion
    schema-generate: NONE
```

sdelamo commented on guides/micronaut-metrics-aws/src/main/resources/application.yml:
```diff
@@ -0,0 +1,58 @@
+    driverClassName: org.h2.Driver
+    username: sa
+    password: ''
+    schema-generate: CREATE_DROP
```
If the app uses flyway, we should not attempt to generate the schema with the data source configuration. This should be `schema-generate: CREATE_DROP`

```suggestion
    schema-generate: NONE
```

sdelamo commented on guides/micronaut-metrics-oci/src/main/resources/application.yml:
```diff
@@ -0,0 +1,58 @@
+  http:
+    services:
+      kucoin: # <1>
+        urls:
```
```suggestion
        url:
```

sdelamo commented on guides/micronaut-metrics-oci/src/test/resources/application-test.yml:
```diff
@@ -0,0 +1,7 @@
+crypto:
+  initialDelay: 10h # <1>
```
```suggestion
  initialDelay: 10h
```

Callout is not shown

sdelamo reviewed (commented):
The screenshots, for example`src/docs/images/oci-metrics/query-editor.png`, use `micronaut_guide` as namespace. The application's name is always in guides `micronautguide`. We recommend the user to use the application name as the namespace name. Can you update that screenshot to say `micronautguide`?

Moroever, we don't explain for AWS if the user needs to deploy the app to ElasticBeanstalk or an EC2 instance to see the metrics, or they should can run it locally after running `aws configure`. Same of Oracle Cloud.

Does it work with GraalVM? If it does, can you please add a section.

n0tl3ss committed:
Update guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc

Co-authored-by: Sergio del Amo <sergio.delamo@softamo.com>

n0tl3ss committed:
Update guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc

Co-authored-by: Sergio del Amo <sergio.delamo@softamo.com>

n0tl3ss committed:
Update guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc

Co-authored-by: Sergio del Amo <sergio.delamo@softamo.com>

n0tl3ss committed:
Update guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc

Co-authored-by: Sergio del Amo <sergio.delamo@softamo.com>

n0tl3ss committed:
Update guides/micronaut-metrics-oci/src/main/resources/application.yml

Co-authored-by: Sergio del Amo <sergio.delamo@softamo.com>

n0tl3ss committed:
Update guides/micronaut-metrics-aws/src/main/resources/application.yml

Co-authored-by: Sergio del Amo <sergio.delamo@softamo.com>

n0tl3ss committed:
Update guides/micronaut-metrics-oci/src/main/resources/application.yml

Co-authored-by: Sergio del Amo <sergio.delamo@softamo.com>

n0tl3ss committed:
Update guides/micronaut-metrics-oci/src/test/resources/application-test.yml

Co-authored-by: Sergio del Amo <sergio.delamo@softamo.com>

n0tl3ss committed:
Update guides/micronaut-metrics-oci/src/main/resources/application.yml

Co-authored-by: Sergio del Amo <sergio.delamo@softamo.com>

n0tl3ss committed:
screnshot update

n0tl3ss commented:
> The screenshots, for example`src/docs/images/oci-metrics/query-editor.png`, use `micronaut_guide` as namespace. The application's name is always in guides `micronautguide`. We recommend the user to use the application name as the namespace name. Can you update that screenshot to say `micronautguide`?
> 
> Moroever, we don't explain for AWS if the user needs to deploy the app to ElasticBeanstalk or an EC2 instance to see the metrics, or they should can run it locally after running `aws configure`. Same of Oracle Cloud.
> 
> Does it work with GraalVM? If it does, can you please add a section.

I am not sure how much we want to repeat ourself. Both aws and oci guide references this guide that `@burtbeckwith`  wrote:

[COLLECT METRICS WITH MICRONAUT](https://guides.micronaut.io/latest/micronaut-metrics-maven-java.html). In that guide the GraalVM is already mentioned. About explaining what is required, is there any example how it was done? In this case notting is required only what you already mention that you have configured SDKs.

sdelamo commented:
> About explaining what is required, is there any example how it was done? In this case notting is required only what you already mention that you have configured SDKs.


In the AWS XRay guide, we tell the users to run the app and then go to the AWS Console: 

https://guides.micronaut.io/latest/micronaut-microservices-distributed-tracing-xray-gradle-java.html#running-the-application

we also tell them to run aws configure

see: 

> 6. AWS Command Line Interface

> Install the [AWS Command Line Interface](https://aws.amazon.com/cli/) and run aws configure. Set the same region you used in the previous sectio

Maybe we could extract the above to a common snippet

In the Lambda guides, we show how to deploy them.

n0tl3ss committed:
Add oracle cloud CLI info

n0tl3ss commented:
> > About explaining what is required, is there any example how it was done? In this case notting is required only what you already mention that you have configured SDKs.
> 
> In the AWS XRay guide, we tell the users to run the app and then go to the AWS Console:
> 
> https://guides.micronaut.io/latest/micronaut-microservices-distributed-tracing-xray-gradle-java.html#running-the-application
> 
> we also tell them to run aws configure
> 
> see:
> 
> > 6. AWS Command Line Interface
> 
> > Install the [AWS Command Line Interface](https://aws.amazon.com/cli/) and run aws configure. Set the same region you used in the previous sectio
> 
> Maybe we could extract the above to a common snippet
> 
> In the Lambda guides, we show how to deploy them.

Something like that already exists in aws guide:

<img width="1237" alt="Screenshot 2022-09-28 at 13 52 09" src="https://user-images.githubusercontent.com/44323106/192772457-4af6f0b2-055f-4e00-bb03-3ee690dcb044.png">

I added requirement for OCI CLI.

graemerocher commented:
@sdelamo is there anything else required to get these guide merged?

sdelamo commented:
I am going to release these guides today.

sdelamo committed:
Merge branch 'master' into guide/oci-metrics

sdelamo committed:
set publication date to today

sdelamo committed:
twitter card for AWS

sdelamo committed:
twitter card for micronaut-metrics-oci

sdelamo committed:
fix: use Amazon CloudWatch not AWS CloudWatch

see: https://aws.amazon.com/cloudwatch/

sdelamo committed:
follow Micronaut guidelines

https://micronaut.io/brand-guidelines/

sdelamo commented:
[I have changed the guides title](https://github.com/micronaut-projects/micronaut-guides/pull/1042/commits/2b9949fc78f5ec3f6c851e371135ffecf1dc4fb7) so that they follow the Micronaut Framework [brand guidelines](https://micronaut.io/brand-guidelines/). 

[I have replaced AWS CloudWatch for Amazon CloudWatch](https://github.com/micronaut-projects/micronaut-guides/pull/1042/commits/17f39be2821966ba9386dae3cbb3be5cc7b23331).  That it is the [name used by Amazon in their website](https://aws.amazon.com/cloudwatch/). 

[I still think we miss to explain how to run the apps](https://github.com/micronaut-projects/micronaut-guides/pull/1042#pullrequestreview-1121832648). We tell the user how to add configuration, and suddenly we show screenshots of the metrics.

sdelamo committed:
change namespace to match copy

In the copy we say:

> Change the namespace to something meaningful, for example, the app’s name. In this guide, we are using micronautguide
